### PR TITLE
xds: allow empty delta update

### PR DESCRIPTION
**Pull request by kkHAIKE**
Commit Message:
when delta xds recv empty repsonce, it should be call *onConfigUpdate*, instead of do nothing, and stop init process 15s (initial_fetch_timeout)

Additional Description:
a first delta xds repsonce can't call wildcard subscription's *onConfigUpdate* (ex: empty CDS), and  stop init process 15s. It don't like SOTW part: https://github.com/envoyproxy/envoy/blob/7f4e35e79ba48e2c6643918e15d0542d881de97b/source/common/config/watch_map.cc#L104-L115
Risk Level: Low 
Testing: Manual Pass
Docs Changes: None
Release Notes: None

### CODE REVIEW

File: source/common/config/watch_map.cc [hunk 1]
```diff
@@ -203,6 +203,12 @@ void WatchMap::onConfigUpdate(
     }
     cur_watch->callbacks_.onConfigUpdate({}, resource_to_remove, system_version_info);
   }
+  // notify empty update
+  if (added_resources.empty() && removed_resources.empty()) {
+    for (auto& cur_watch : wildcard_watches_) {
+      cur_watch->callbacks_.onConfigUpdate({}, {}, system_version_info);
+    }
+  }
```
Comment (htuch) [inline source/common/config/watch_map.cc:211]: Looks good, can you add a regression test? Thanks.
/wait

**Comment by kkHAIKE** [comment]
ping @htuch :smile:

File: test/common/config/new_grpc_mux_impl_test.cc [hunk 2]
```diff
@@ -90,7 +90,8 @@ TEST_F(NewGrpcMuxImplTest, DiscoveryResponseNonexistentSub) {
         std::make_unique<envoy::service::discovery::v3::DeltaDiscoveryResponse>();
     unexpected_response->set_type_url(type_url);
     unexpected_response->set_system_version_info("0");
-    EXPECT_CALL(callbacks_, onConfigUpdate(_, _, "0")).Times(0);
+    // empty response should call onConfigUpdate on wildcard watch
+    EXPECT_CALL(callbacks_, onConfigUpdate(_, _, "0")).Times(1);
```
Comment (htuch) [inline test/common/config/new_grpc_mux_impl_test.cc:94]: Nit: `Times(1)` is redundant, you can just drop that and have the `EXPECT_CALL`.

**Review by htuch — APPROVED** [review]

**Comment by htuch** [comment]
LGTM, thanks!

**Pull request merged**